staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -1302,16 +1302,6 @@ func (ca *ConnectionAgent) remoteAttInVNWithVirtualIPHostIPSelector(vni uint32)
 	// remote node.
 	remoteAtt := k8sfields.OneTermNotEqualSelector(attNode, ca.localNodeName)
 
-	// hostIPIsNotLocal expresses the constraint that the NetworkAttachment
-	// status.hostIP is not equal to that of the IP of the connection agent.
-	// Without this selector, an update to the spec.Node field of a NetworkAttachment
-	// could lead to a creation notification for the attachment in a remote
-	// attachments cache, even if the attachment still has the host IP of the
-	// connection node (status.hostIP is set with an update by the connection
-	// agent on the node of the attachment). This could result in the creation
-	// of a remote interface with the host IP of the connection agent (local) node.
-	hostIPIsNotLocal := k8sfields.OneTermNotEqualSelector(attHostIP, ca.hostIP.String())
```

Comment:
This selector was needed because if spec.Node was updated causing an attachment to transition from local to remote, the fact that status.hostIP still stored the local hostIP caused transients during which a remote attachment had a remote interface with a local host IP.  With immutability this is not needed, an attachment’s node can change only through a delete and a re-create, which will reset the status, including hostIP.